### PR TITLE
chore: Bumping version of the Terraform provider

### PR DESCRIPTION
**Pull request by Gmerold**
# Description

Bumps version of the Terraform provider to `0.11.0`

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have bumped the version of the library

### CODE REVIEW

File: modules/sdcore-control-plane-k8s/terraform.tf [hunk 1]
```diff
@@ -5,7 +5,7 @@ terraform {
   required_providers {
     juju = {
       source  = "juju/juju"
-      version = "~> 0.10.1"
+      version = ">= 0.10.1"
```
Comment (gatici) [inline modules/sdcore-control-plane-k8s/terraform.tf:8]: Could we write  ~> 0.11.0 ?

File: modules/sdcore-user-plane-k8s/terraform.tf [hunk 2]
```diff
@@ -5,7 +5,7 @@ terraform {
   required_providers {
     juju = {
       source  = "juju/juju"
-      version = "~> 0.10.1"
+      version = ">= 0.10.1"
```
Comment (gatici) [inline modules/sdcore-user-plane-k8s/terraform.tf:8]: Could we write ~> 0.11.0 ?

File: modules/sdcore-control-plane-k8s/terraform.tf (hunk 1 above)
Comment (Gmerold) [reply modules/sdcore-control-plane-k8s/terraform.tf:8]: Ups, it should be `~> 0.11.0`... Looks like I accidentally pushed my test change...

Comment (Gmerold) [reply modules/sdcore-control-plane-k8s/terraform.tf]: Fixed.

**Review by gatici — APPROVED** [review]

**Pull request merged**